stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -195,7 +204,12 @@ protected synchronized void onFirstPeerRegistered() {
 
     @Override
     protected synchronized void onLastPeerUnregistered() {
-      mObjectIdMapper.clear();
+      mDOMProvider.postAndWait(new Runnable() {
+        @Override
+        public void run() {
+          mObjectIdMapper.clear();
+        }
+      });
 
       mDOMProvider.dispose();
```

Comment:
Which part? Why wouldn't it be? ObjectIdMapper locks internally. clear() is implemented carefully to make sure that this is a plausible way to do things (it doesn't lock while broadcasting the onUnmapped calls).